src/components/InputV2/Base.tsx

```diff
@@ -1,6 +1,61 @@
+import styled, { css } from 'styled-components'
 import PropTypes from 'prop-types'
 import { Label } from './Label'
 import { IconClose } from '../Icons'
+import { space } from '../theme'
+
+const BaseLabel = styled(Label)`
+  .button-wrapper {
+    display: flex;
+    align-items: center;
+    gap: ${space()};
+
+    > *:first-child {
+      flex: 1 0 auto;
+    }
+
+    > *:not(:first-child) {
+      flex: 0 0 auto;
+      transition: transform 0.1s ease-out;
+
+      &:hover:not(:active) {
+        transform: scale(1.2);
+      }
+
+      &:active {
+        transition: none;
+      }
+    }
+  }
+
+  .unit-wrapper {
+    position: relative;
+
+    input {
+      width: 100%;
+    }
+
+    .unit {
+      position: absolute;
+      top: 50%;
+      right: 1em;
+      transform: translateY(-50%);
+      color: var(--purple-medium);
+
+      ${props =>
+        props.error &&
+        css`
+          color: var(--red-dark) !important;
+        `}
+
+      ${props =>
+        props.disabled &&
+        css`
+          color: var(--gray-dark) !important;
+        `}
+    }
+  }
+`
```

Comment:
@Schwartz10 For all of the inputs that have specific styling requirements in `Label` (like `.button-wrapper`, `.unit-wrapper`), I created their own specific `Label` extension. So the styling definition is now closer to the HTML where they apply, which is nice. But also the styles won't be applied to elements that don't need it. You can see by the classes that don't apply to specific elements being removed from the snapshots

Reply:
Good thinking!